### PR TITLE
[Mac] #769 Added error reporting via Fabric/Crashlytics

### LINKED ISSUE #769: [Mac] Add error-reporting capability

### PR DESCRIPTION
**Pull request by tombogle**
Fixes #769

### CODE REVIEW

File: mac/build.sh [hunk 1]
```diff
@@ -270,9 +270,15 @@ updatePlist() {
...
-			echo "Setting $KM_COMPONENT_NAME version to $KM_VERSION"
+			echo "Setting $KM_COMPONENT_NAME version to $KM_VERSION in $KM_PLIST"
 			/usr/libexec/Plistbuddy -c "Set CFBundleVersion $KM_VERSION" "$KM_PLIST"
 			/usr/libexec/Plistbuddy -c "Set CFBundleShortVersionString $KM_VERSION" "$KM_PLIST"
+			if [ "$CONFIG" == "Release" ]; then
+				echo "Setting Fabric APIKey for release build in $KM_PLIST"
+				if [ "$FABRIC_API_KEY_KEYMAN4MACIM" == "" ]
+				    fail "FABRIC_API_KEY_KEYMAN4MACIM environment variable not set!"
+				/usr/libexec/Plistbuddy -c "Set Fabric:APIKey $FABRIC_API_KEY_KEYMAN4MACIM" "$KM_PLIST"
+			fi
```
Comment (darcywong00) [inline mac/build.sh]: The CI server says you're missing an `fi`

File: mac/build.sh [hunk 2]
```diff
@@ -270,9 +270,15 @@ updatePlist() {
 	    KM_COMPONENT_NAME="$2"
 		KM_PLIST="$KM_COMPONENT_BASE_PATH/$KM_COMPONENT_NAME/Info.plist"
 		if [ -f "$KM_PLIST" ]; then 
-			echo "Setting $KM_COMPONENT_NAME version to $KM_VERSION"
+			echo "Setting $KM_COMPONENT_NAME version to $KM_VERSION in $KM_PLIST"
 			/usr/libexec/Plistbuddy -c "Set CFBundleVersion $KM_VERSION" "$KM_PLIST"
 			/usr/libexec/Plistbuddy -c "Set CFBundleShortVersionString $KM_VERSION" "$KM_PLIST"
+			if [ "$CONFIG" == "Release" ]; then
+				echo "Setting Fabric APIKey for release build in $KM_PLIST"
+				if [ "$FABRIC_API_KEY_KEYMAN4MACIM" == "" ]
```
Comment (darcywong00) [inline mac/build.sh]: Should this be `if [ ... ]; then`?

File: mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodAppDelegate.m [hunk 3]
```diff
@@ -96,6 +98,11 @@ - (id)init {
     return self;
 }
 
+-(void)applicationDidFinishLaunching:(NSNotification *)aNotification {
+    [[NSUserDefaults standardUserDefaults] registerDefaults:@{ @"NSApplicationCrashOnExceptions": @YES }];
```
Comment (darcywong00) [inline mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodAppDelegate.m]: 1. Is this the only place you have Keyman sending Crashlytics?
2. Is it possible to track new keyboard installations? (Either in this PR or a future one)

Comment (mcdurdin) [reply mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodAppDelegate.m]: Additional analytics can wait until 11.0. Crash reporting is important for managing reliability; everything else is data to help us prioritise (and we have plenty to do now anyway without needing more data!)

**Pull request merged**